pennylane/transforms/specs.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Code for resource estimation"""
+
+
+def specs(qnode, max_expansion=None):
+    """
+
+    Args:
+        qnode (.QNode): the QNode to calculation the specifications for
+
+    Keyword Args:
+        max_expansion=None (int): The number of times the internal circuit should be expanded when
+            executed on a device. Expansion occurs when an operation or measurement is not
+            supported, and results in a gate decomposition. If any operations in the decomposition
+            remain unsupported by the device, another expansion occurs. Defaults to
+            ``qnode.max_expansion``.
```

Comment:
Just to understand, why do we give access to exactly that one? For example, I would find it much more useful to give user access to the stopping criterion for expansion, if anything...

Reply:
I'm just mirroring how QNode's work.

Reply:
I also wondered :thinking:. If we have to expand to depth `N` for the device to execute, what would happen if `max_expansion < N`? Wouldn't it just not work on the device? Although I guess you can still access the specs?

Reply:
E.g.,
```python
dev = qml.device('default.qubit', wires=2)
@qml.qnode(dev)
def circuit():
    qml.templates.StronglyEntanglingLayers(np.random.random((3, 2, 3)), wires=range(2))
    return qml.probs(wires=(0,1))

qml.specs(circuit, max_expansion=0)()
```